CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc

```diff
@@ -0,0 +1,245 @@
+// -*- C++ -*-
+//
+// Package:    CondFormats/PCLConfig
+// Class:      AlignPCLThresholdsWriter
+// 
+/**\class AlignPCLThresholdsWriter AlignPCLThresholdsWriter.cc CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
+
+ Description: class to build the SiPixelAli PCL thresholds
+
+*/
+//
+// Original Author:  Marco Musich
+//         Created:  Wed, 22 Feb 2017 12:04:36 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/PCLConfig/interface/AlignPCLThresholds.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+//
+// class declaration
+//
+
+namespace DOFs {
+  enum dof{
+    X,
+    Y,
+    Z,
+    thetaX,
+    thetaY,
+    thetaZ,
+    extraDOF
+  };
+}
+
+class AlignPCLThresholdsWriter : public edm::one::EDAnalyzer<>  {
+   public:
+      explicit AlignPCLThresholdsWriter(const edm::ParameterSet&);
+      ~AlignPCLThresholdsWriter();
+
+      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+   private:
+      virtual void beginJob() override;
+      virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+      virtual void endJob() override;
+      DOFs::dof mapOntoEnum(std::string coord);
+
+      // ----------member data ---------------------------
+      const std::string m_record;
+      const unsigned int m_minNrecords;
+      const std::vector<edm::ParameterSet> m_parameters;
+};
+
+//
+// constructors and destructor
+//
+AlignPCLThresholdsWriter::AlignPCLThresholdsWriter(const edm::ParameterSet& iConfig):
+  m_record(iConfig.getParameter<std::string>("record")),
+  m_minNrecords(iConfig.getParameter<unsigned int>("minNRecords")),
+  m_parameters(iConfig.getParameter<std::vector<edm::ParameterSet> >("thresholds"))
+{
+  //now do what ever initialization is needed
+}
+
+
+AlignPCLThresholdsWriter::~AlignPCLThresholdsWriter()
+{
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+AlignPCLThresholdsWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+   using namespace edm;
+  
+   AlignPCLThresholds* myThresholds = new AlignPCLThresholds();
```

Comment:
@ghellwig thanks for the tip. The following patch 
```
diff --git a/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc b/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
index 67e4f35..d7fb1eb 100644
--- a/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
diff --git a/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc b/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
index 67e4f35..d7fb1eb 100644
--- a/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
+++ b/CondFormats/PCLConfig/plugins/AlignPCLThresholdsWriter.cc
@@ -63,8 +63,6 @@ class AlignPCLThresholdsWriter : public edm::one::EDAnalyzer<>  {
       const std::string m_record;
       const unsigned int m_minNrecords;
       const std::vector<edm::ParameterSet> m_parameters;
-      AlignPCLThresholds* myThresholds;
-
 };
 
 //
@@ -76,13 +74,11 @@ AlignPCLThresholdsWriter::AlignPCLThresholdsWriter(const edm::ParameterSet& iCon
   m_parameters(iConfig.getParameter<std::vector<edm::ParameterSet> >("thresholds"))
 {
   //now do what ever initialization is needed
-  myThresholds = new AlignPCLThresholds();
 }
 
 
 AlignPCLThresholdsWriter::~AlignPCLThresholdsWriter()
 {
-  delete myThresholds;
 }
 
 //
@@ -94,8 +90,9 @@ void
 AlignPCLThresholdsWriter::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
 {
    using namespace edm;
-  
-   edm::LogInfo("AlignPCLThresholdsWriter")<<"Size of AlignPCLThresholds object "<< myThresholds->size() <<std::endl<<std::endl;
+
+   std::unique_ptr<AlignPCLThresholds> myThresholds = std::unique_ptr<AlignPCLThresholds>(new AlignPCLThresholds());
+   edm::LogInfo("AlignPCLThresholdsWriter")<<"Size of AlignPCLThresholds object "<< myThresholds.get()->size() <<std::endl<<std::endl;
    
    // loop on the PSet and insert the conditions 
 
@@ -167,13 +164,13 @@ AlignPCLThresholdsWriter::analyze(const edm::Event& iEvent, const edm::EventSetu
           }
         
         AlignPCLThreshold a(my_X,my_tX,my_Y,my_tY,my_Z,my_tZ,extraDOFs);
-        myThresholds->setAlignPCLThreshold(alignableId,a);
+        myThresholds.get()->setAlignPCLThreshold(alignableId,a);
         
        } // if alignable is found in the PSet
      } // loop on the PSets
 
      // checks if all mandatories are present
-     edm::LogInfo("AlignPCLThresholdsWriter")<<"Size of AlignPCLThresholds object  "<<myThresholds->size() <<std::endl;
+     edm::LogInfo("AlignPCLThresholdsWriter")<<"Size of AlignPCLThresholds object  "<<myThresholds.get()->size() <<std::endl;
      for(auto& mandatory : mandatories){
        if(std::find(presentDOF.begin(), presentDOF.end(), mandatory) == presentDOF.end()) {
         edm::LogWarning("AlignPCLThresholdsWriter")<<"Configuration for DOF: "<<mandatory<<" for alignable "<< alignable <<"is not present \n" 
@@ -184,18 +181,18 @@ AlignPCLThresholdsWriter::analyze(const edm::Event& iEvent, const edm::EventSetu
    } // ends loop on the alignable units
    
    // set the minimum number of records to be used in pede
-   myThresholds->setNRecords(m_minNrecords);
+   myThresholds.get()->setNRecords(m_minNrecords);
    edm::LogInfo("AlignPCLThresholdsWriter")<<"Content of AlignPCLThresholds "<<std::endl;
 
    // use buil-in method in the CondFormat
-   myThresholds->printAll();
+   myThresholds.get()->printAll();
 
    // Form the data here
    edm::Service<cond::service::PoolDBOutputService> poolDbService;
    if( poolDbService.isAvailable() ){
      cond::Time_t valid_time = poolDbService->currentTime(); 
      // this writes the payload to begin in current run defined in cfg
-     poolDbService->writeOne(myThresholds,valid_time, m_record);
+     poolDbService->writeOne(myThresholds.get(),valid_time, m_record);
    }
 }
```
works OK. 
I don't have preferences. If there is preference towards the `unique_ptr` I'll update the branch.